### ISSUE TITLE
Questions about fixed frames

### ISSUE THREAD

**Issue by Luwang012** [jkk-research/urban_road_filter#19]
hello,i encountered some issues while running the code.when i try to use the rslidar and change the fixed frame in the LidarFilter.cfg,but i find that it dosen't work normal although i recompile the code.Is this code only applicable to the velydone or ouster?Can u give me something suggestion?thank u.

**Comment by horverno** [comment]
hello @Luwang012 don't change it in the `LidarFilters.cfg`, its just a config for rqt_reconfigure. Use a launch file similar to this:


``` xml
<launch>
    <node name="rqt_conf_gui" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen"/>
    <node name="rviz_config" pkg="rviz" type="rviz"  args="-d $(find urban_road_filter)/config/demo1.rviz" output="screen"/>
    <group ns="urban_road_filter">
        <node name="urban_road_filt" pkg="urban_road_filter" type="lidar_road" output="screen">
            <param name="topic_name" value="/rslidar/or_other_lidar/points"/>
            <param name="fixed_frame" value="rslidar/or_other_lidar_frame"/>
        </node>
    </group>
</launch>
```

Change the `rslidar/or_other_lidar` of course.

**Comment by Luwang012** [comment]
Thank you for your answer！

**Comment by enginksz** [comment]
Hello,

First of all, thank you for the project.

When I tested it with my rosbag record, the screen freezes after 2-3 seconds in rviz and gives an error as in the attachment.

I ran commands like "rosdep update". I could also run it in the rosbag record you gave as an example, I did not get any error.

What could be the reason for this freeze in rviz? Could it be that it gives an error when it cannot find a road ? 

![Screenshot from 2024-10-31 15-51-41](https://github.com/user-attachments/assets/1650ff9c-cf19-46ce-b371-10c3251e215f)
